### PR TITLE
Conditional Processor rewrite to not use new Function()

### PR DESCRIPTION
**Pull request by cdloh**
They are not allowed in the Cloudflare worker runtime

Instead split condition strings out into its components and evaluate it ourselves

### CODE REVIEW

File: src/processConditionals.ts [hunk 1]
```diff
@@ -131,102 +136,214 @@ function esi_eval_var_in_when_tag(
...
-    "|": " || ",
-    "&": " && ",
-    "||": " || ",
-    "&&": " && ",
-    "!": " ! ",
+    "|": "||",
+    "&": "&&",
+    "||": "||",
+    "&&": "&&",
+    "!": "!",
```
Comment (asharpe-squiz) [inline src/processConditionals.ts:152]: This replacement seems unnecessary!?

Comment (cdloh) [reply src/processConditionals.ts:152]: Good point isn't needed anymore.

**Comment by jrencz** [comment]
@cdloh can you please consider publishing a version that would include this PR?

The latest (0.1.2) does not and I just bumped into the problem which you solved

(BTW: great piece of work! I appreciate)

**Comment by cdloh** [comment]
@jrencz I've been meaning to get some proper automated versioning done

But all commits to main do get a published version it's just a 0.0.0 version with the git SHA afterwards

0.0.0-1f7bea36fa360e1719ee2fff79fef2187e407cfb

Is the current main branch. Install that for the mean time. I'll try and get some time in the coming weeks to get some proper versions published 

Love to hear how you go with it / what you're using it for :)

**Comment by jrencz** [comment]
@cdloh I've had Varnish doing ESI for a long time on main domain. Right now I try doing same ESI trickery for domains that are orange-clouded and for Cloudflare Pages - Worker is the greatest common denominator for all those. Plus: I will be able to dismiss Varnish as surrogate

**Comment by cdloh** [comment]
@jrencz I've published 0.2.0 which has the above fix along with some others.

Let me know if you have any issues :)

**Comment by jrencz** [comment]
I gave it a go, and **so far so good** - conditional processing works. Thank you for a quick reaction

**Pull request merged**